components/usage/pkg/db/cost_center.go

```diff
@@ -263,3 +287,54 @@ func (c *CostCenterManager) ListLatestCostCentersWithBillingTimeBefore(ctx conte
 
 	return results, nil
 }
+
+func (c *CostCenterManager) ResetUsage(ctx context.Context, cc CostCenter) (CostCenter, error) {
+	if cc.BillingStrategy != CostCenter_Other {
+		return CostCenter{}, fmt.Errorf("cannot reset usage for Billing Strategy %s for Cost Center ID: %s", cc.BillingStrategy, cc.ID)
+	}
+
+	entity, _ := cc.ID.Values()
+
+	// We do not carry over the spending limit from the existing CostCenter.
+	// At the moment, we don't have a use case for it. Getting the spending limit from configured values
+	// ensures that we progressively update the spending limit to configured values rather than having to
+	// perform bulk DB queries when the defaults do change.
+	var spendingLimit int32
+	switch entity {
+	case AttributionEntity_Team:
+		spendingLimit = c.cfg.ForTeams
+	case AttributionEntity_User:
+		spendingLimit = c.cfg.ForUsers
+	default:
+		return CostCenter{}, fmt.Errorf("cannot reset usage for unknown attribution entity ID: %s", cc.ID)
+	}
+
+	now := time.Now().UTC()
+
+	// Default to 1 month from now, if there's no nextBillingTime set on the record.
```

Comment:
Do I get this right: If the previous billing cycle ended already, we start the new one _now_ instead of based on the old one + iterating months until now?

Just asking because conceptually, the later might be simpler to grasp mentally and thus might be easier to debug and change later on.

Reply:
This is just a safety net for the case where no NextBillingTime was present on the record. If it is present, we do (old.NextBillingTime + 1 month). Otherwise, we do (now + 1 month)

Reply:
> f it is present, we do (old.NextBillingTime + 1 month). Otherwise, we do (now + 1 month)

And if there is a gap of say 3 months? Where do we go from old.nextBillingTime to sth that covers today? :thinking:

Reply:
There's a periodic job which triggers the reset for expired every 15 minutes. This should cover most cases.

In the case it the CostCenter expires between now and (-15 minutes), we reset when we're fetching the CostCenter.

In the rare case where we've been failing to reset for a while, the reset catches up by incrementally (in each iteration) moving the BillingTime forward by 1 month.

Reply:
> There's a periodic job which triggers the reset for expired every 15 minutes.

Great, solved. :heavy_check_mark: :lotus_position: